src/components/search-by-person-attributes/search-by-person-attributes.component.tsx

```diff
@@ -0,0 +1,122 @@
+import React, {
+  Dispatch,
+  SetStateAction,
+  useCallback,
+  useEffect,
+  useState,
+} from "react";
+
+import { showNotification } from "@openmrs/esm-framework";
+import { Column, Dropdown, TextInput } from "carbon-components-react";
+import { useTranslation } from "react-i18next";
+
+import { PersonAttribute, SearchParams } from "../../types";
+import { getPersonAttributes } from "./search-by-person-attributes.resource";
+import styles from "./search-by-person-attributes.style.scss";
+import {
+  getQueryDetails,
+  getSearchByAttributesDescription,
+} from "./search-by-person-attributes.utils";
+
+interface SearchByPersonAttributesProps {
+  setQueryDescription: Dispatch<SetStateAction<String>>;
+  setSearchParams: Dispatch<SetStateAction<SearchParams>>;
```

Comment:
I don't think you should pass parent's set state function here. Just pass a callback function to make things more readable. Let's go through an on-call code review. I have some doubts to clarify.